include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -278,15 +281,15 @@ class IExecutionProvider {
 
   // helper to generate ids that are unique to model and deterministic, even if the execution provider is shared across
   // multiple sessions.
-  class ModelMetadefIdGenerator {
+  class ModelIdGenerator {
    public:
-    int GenerateId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash);
+    int GenerateId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash, bool hash_model_name = false, bool hash_nodes = false);
 
```

Comment:
hash_model_name needs to default to true to preserve current behavior does it not?

Reply:
actually I see there's some new logic differentiating between the full path and just the name. this should be called out in documentation of the function given it's not implicitly obvious.

Reply:
comments are added